src/DIRAC/Core/Utilities/TimeUtilities.py

```diff
@@ -99,6 +107,9 @@ def fromEpoch(epoch):
     """
     Get datetime object from epoch
     """
+    # Check if the timestamp is in milliseconds
+    if len(str(epoch)) == 13:
```

Comment:
Mmm... I don't think I like it enormously

Reply:
I think we should at least do this which gives us from 1973 to 5138:
```suggestion
    if epoch > 10**17:  # nanoseconds
        epoch /= 1000**3
    elif epoch > 10**14:  # microseconds
        epoch /= 1000**2
    elif epoch > 10**11:  # milliseconds
        epoch /= 1000
```

Please add unit tests to make sure that `datetime(5100, 1, 1).timestamp()` and `datetime(1974, 1, 1).timestamp()` are always converted correctly even when multiplied by 1000, 1000000 and 1000000000.